spec/unit/functions/lookup_spec.rb

```diff
@@ -2304,6 +2306,19 @@ def lookup(key, scope, order_override, resolution_type, context)
             expect(notices).to eql(['success'])
           end
         end
+
+        context 'with declared but incompatible return_type' do
+          let(:puppet_function) { <<-PUPPET.unindent }
+            function mod_a::pp_lookup_key(Puppet::LookupKey $key, Hash[String,String] $options, Puppet::LookupContext $context) >> Runtime['ruby','Symbol'] {
+              undef
+            }
+            PUPPET
+
+          it 'fails and reports error' do
+            expect{lookup('mod_a::a')}.to raise_error(
+              "Return type of 'lookup_key' function named 'mod_a::pp_lookup_key' is incorrect, expects a value of type Scalar, Sensitive, Type, Hash, or Array, got Runtime")
```

Comment:
I added [PUP-7371](https://tickets.puppetlabs.com/browse/PUP-7371) as a blocker for this ticket.